src/components/serializedChallenge.ts

```diff
@@ -0,0 +1,100 @@
+import simpleTypeGuard, { SimpleArray, SimpleStringOptional, SimpleNumber, SimpleString, SimpleBooleanOptional, SimpleObjectOptional, SimpleSkip } from 'simple-type-guard';
+
+
+export type SerializedChallenge = {
+    fileVersion: number,
+    title: string,
+    statement: {
+        description: string,
+        clue?: string
+    },
+    scene: Scene,
+    toolbox: {
+        blocks: string[], // for now, block ids, future: could be objects.
+        categorized?: boolean // default true
+    },
+    stepByStep?: boolean, // default false
+    predefinedSolution?: string,
+    assesments?: Assesments
+}
+
+
+const sceneTypes =  ["lita", "duba", "toto", "coty", "niandu", "pinguine", "yaguarete", "carpincho", "custom"] as const //Used for file validity checking
+type SceneType = typeof sceneTypes[number]
+
+const cells = ["P", "-", "A", ""] as const //Used for file validity checking. Should consider different objects / empty cell / no cell at all
+type Cell = typeof cells[number]
+
+type SceneMap = Cell[]
+
+export type Scene = {
+    type: SceneType
+    maps: SceneMap[]
+}
+
+type Assesments =  {
+    itWorks?: boolean, // old "debeFelicitar", default true
+    decomposition?: DecompositionAssessment,
+    simpleRepetition?: boolean,
+    conditionalRepetition?: boolean,
+    conditionalAlternative?: boolean,
+}
+
+type DecompositionAssessment = { maxProgramLength: number }
+
+
+export const isValidChallenge = (json: unknown): json is SerializedChallenge => {
+    const structureIsValid: boolean = simpleTypeGuard<SerializedChallenge>(json, {
+        fileVersion: SimpleNumber,
+        title: SimpleString,
+        statement: {description: SimpleString, clue: SimpleStringOptional},
+        toolbox: {blocks: new SimpleArray(SimpleString), categorized: SimpleBooleanOptional},
+        stepByStep: SimpleBooleanOptional,
+        predefinedSolution: SimpleStringOptional,
+        scene: SimpleSkip, //We cant check if the scene values are valid here, so we need to check them manually. This is because "SimpleExactMatch" seems to be broken.
+        assesments: new SimpleObjectOptional<Assesments>({
```

Comment:
El problema con la escena es que tiene valores concretos que son validos. Esto se deberia solucionar con el `SimpleExactMatch` de esta dependencia que usamos:
![image](https://user-images.githubusercontent.com/37090248/233133686-dd727d21-8a9c-47c5-ad81-53077f6f326c.png)

El problema es que parece no andar 👍. Al solo usar esta función typescript marca su uso como un error diciendo que "Type instantiation is excessively deep and possibly infinite". Esto puede ser porque typescript esta haciendo mucho procesamiento porque el tipo es muy complejo o porque hay algo recursivo en el medio. Es posible que sea por algun error de la dependencia que lleva a que haya algo recursivo y que no sea debido a que nuestro tipo es muy complejo, porque probando el caso mas simple de esta funcion, osea algo como `simpleTypeGuard('red', new SimpleExactMatch(['red'])`, sigue tirando este error. 

Por esto, el tipo de scene lo checkeo manualmente.